### ISSUE TITLE
Using release-it programmatically -> Error [ERR_REQUIRE_ESM]

### ISSUE THREAD

**Issue by borsTiHD** [release-it/release-it#775]
Hi all,

first I want to thank you for this project.
I have not been able to use it yet, but wanted to implement it in a project and I am already looking forward to it.

However I would like to use it in my small build script. [(build.js)](https://github.com/borsTiHD/pi-control/blob/main/src/build.js)
My problem is, I'm using ESM syntax and I don't want to go back to commonJS.
Is there a way to use it that way?

With kind regards
borsTiHD

**Comment by webpro** [comment]
> My problem is, I'm using ESM syntax and I don't want to go back to commonJS.
Is there a way to use it that way?

If this doesn't work yet I want to start supporting it. What is the issue and how can we fix it?

**Comment by borsTiHD** [comment]
I use it now for the moment as you recommend it in your docs (script in package.json).
Was working like a charm and I will use `release-it` forever now. :)

Back to my wish. :D
In principle, my build script can be broken down to the following parts.
You can see the hole thing in my linked repo from above (just commented out).

package.json:
```
{
  ...
  "type": "module",
}
```

build.js:
```
import release from 'release-it'
import releaseItConfig from '../.release-it.js'

const options = releaseItConfig
return release(options).then((output) => {
    console.log(output) // { version, latestVersion, name, changelog }
})
```

.release-it.js:
```
export default {
    git: {
        commitMessage: 'chore(release): 🚀 v${version}',
        requireBranch: 'main', // Stopps if your current branch not match
        tag: true,
        tagName: 'v${version}',
        tagAnnotation: 'Release v${version}',
    },
    github: {
        release: true,
        releaseName: 'v${version}',
        assets: ['build/*.tar.gz']
    },
    npm: {
        publish: false
    }
}
```

Output:
```
Error [ERR_REQUIRE_ESM]: Must use import to load ES Module: D:\Development\Workspace\pi-control\.release-it.js
require() of ES modules is not supported.
require() of D:\Development\Workspace\pi-control\.release-it.js from D:\Development\Workspace\pi-control\node_modules\release-it\node_modules\cosmiconfig\dist\loaders.js is an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which defines all .js files in that package scope as ES modules.
Instead rename .release-it.js to end in .cjs, change the requiring code to use import(), or remove "type": "module" from D:\Development\Workspace\pi-control\package.json.

    at new NodeError (node:internal/errors:363:5)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1126:13)
    at Module.load (node:internal/modules/cjs/loader:989:32)
    at Function.Module._load (node:internal/modules/cjs/loader:829:14)
    at Module.require (node:internal/modules/cjs/loader:1013:19)
    at module.exports (D:\Development\Workspace\pi-control\node_modules\import-fresh\index.js:32:59)
    at loadJs (D:\Development\Workspace\pi-control\node_modules\release-it\node_modules\cosmiconfig\dist\loaders.js:16:18)
    at ExplorerSync.loadFileContentSync (D:\Development\Workspace\pi-control\node_modules\release-it\node_modules\cosmiconfig\dist\ExplorerSync.js:84:26)
    at ExplorerSync.createCosmiconfigResultSync (D:\Development\Workspace\pi-control\node_modules\release-it\node_modules\cosmiconfig\dist\ExplorerSync.js:89:30)
    at ExplorerSync.loadSearchPlaceSync (D:\Development\Workspace\pi-control\node_modules\release-it\node_modules\cosmiconfig\dist\ExplorerSync.js:70:25)
    at ExplorerSync.searchDirectorySync (D:\Development\Workspace\pi-control\node_modules\release-it\node_modules\cosmiconfig\dist\ExplorerSync.js:55:32)
    at run (D:\Development\Workspace\pi-control\node_modules\release-it\node_modules\cosmiconfig\dist\ExplorerSync.js:35:27)
    at cacheWrapperSync (D:\Development\Workspace\pi-control\node_modules\release-it\node_modules\cosmiconfig\dist\cacheWrapper.js:28:18)
    at ExplorerSync.searchFromDirectorySync (D:\Development\Workspace\pi-control\node_modules\release-it\node_modules\cosmiconfig\dist\ExplorerSync.js:47:49)
    at ExplorerSync.searchSync (D:\Development\Workspace\pi-control\node_modules\release-it\node_modules\cosmiconfig\dist\ExplorerSync.js:27:25)
    at getLocalConfig (D:\Development\Workspace\pi-control\node_modules\release-it\lib\config.js:34:78) {
  code: 'ERR_REQUIRE_ESM'
}
```

Regarding the how, unfortunately, I can not help. I only know that there are possibilities to publish npm modules in both variants, but I don't know the effort for that. I am only a small hobby developer.

**Comment by webpro** [comment]
> Instead rename .release-it.js to end in .cjs

Looks like a conversion of `.release-it.js` to `.release-it.cjs` or `.release-it.json` could fix the issue?

**Comment by borsTiHD** [comment]
Oh I found my problem. Its working with ESM syntax.
I renamed my release-it config in something different: `.release-it-config.js` (still written in `export default`)
Now its working with the snippets I postet above.

Could it be, that release-it trys to require() `.release-it.js` even if I put my own importet `.release-it.js` as an config argument? Practically like a double import?

With a different name and importet as an argument it works perfetcly. ❤️
Thank you so much again for this project and your help.
